src/components/HypothesisBuilder/components/HypothesisTree/HypothesisRow.js

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import PropTypes from "prop-types";
+import cx from "clsx";
+import { Icon } from "../../../Icon";
+
+export const HypothesisRow = ({
+  renderChevronIcon,
+  iconLeft,
+  iconLeftClassName,
+  text,
+  image,
+  onOpenModal,
+  showMessageIcon,
+  showSignalIcon
+}) => (
+  <div className="w-full flex justify-between rounded-8px border focus:outline-none bg-white">
+    <div className="flex items-center h-12 w-full">
+      {renderChevronIcon()}
+      <div
+        className={cx(
+          "ml-2 bg-lighter-gray rounded-full p-2 w-8 h-8",
+          iconLeftClassName
+        )}
+      >
+        {iconLeft}
+      </div>
+      <span
+        className="ml-3 text-sm font-normal w-full text-left"
+        onClick={onOpenModal}
+        role="presentation"
+      >
+        {text}
+      </span>
+    </div>
+    <div className="flex items-center h-12 pr-5 ml-1.5">
+      {showMessageIcon && <Icon name="message" className="ml-3.5 mr-2.5" />}
+      {showSignalIcon && <Icon name="diagram" />}
+      {/(http)?s?:?(\/\/[^"']*\.(?:png|jpg|jpeg|gif|svg))/g.test(image) ? (
+        <img
+          className="h-6 w-6 ml-4 mr-3.5 rounded-full bg-gray-400 flex items-center justify-center"
+          src={image}
+          alt={image}
+        />
+      ) : (
+        <div className="h-6 w-6 ml-4 mr-3.5 rounded-full bg-gray-400 flex items-center justify-center text-2xs">
+          {image !== null
+            ? image
```

Comment:
What does this do ?

Reply:
<img width="529" alt="Screenshot 2022-02-04 at 3 39 11 PM" src="https://user-images.githubusercontent.com/32077311/152538577-af16a64b-3bac-4342-b9ac-5e5c0895e260.png">
It's for an avatar. Check if the image is a valid image link, if yes then display an avatar. If not then display a circle with initials (if name not empty other case display empty circle)

Reply:
Then it should move to the Avatar component with an appropriate function name

Reply:
@moshloop do you mind if we install react-image package ?